GMG/GMG/Scenes/Recording/RecordingIntent.swift

```diff
@@ -0,0 +1,160 @@
+//  Copyright © 2025 ADA 4th GMG. All rights reserved.
+
+import Combine
+import Foundation
+
+protocol RecordingIntentProtocol {
+    func onTapRecordButton()
+    func onTapStopRecordButton()
+    func onTapResetButton()
+    func onTapPlayButton(_ url: URL)
+    func onTapStopPlayButton()
+    func onTapNextButton(_ url: URL, completion: @escaping () -> Void)
+}
+
+final class RecordingIntent: RecordingIntentProtocol {
+    private weak var model: RecordingModelActionProtocol?
+
+    private let recordManager: RecordManager
+    private let playbackManager: PlaybackManager
+
+    private let scoreFactory: ScoreFactory
+
+    private var cancellables: Set<AnyCancellable>
+
+    private var scoreCreationTask: Task<Void, Never>?
+
+    init(model: RecordingModelActionProtocol) {
+        self.model = model
+
+        self.recordManager = RecordManager()
+        self.playbackManager = PlaybackManager()
+
+        self.scoreFactory = ScoreFactory()
+
+        self.cancellables = Set<AnyCancellable>()
+
+        self.scoreCreationTask = nil
+
+        setupPublishers()
+    }
+
+    private func setupPublishers() {
+        recordManager.isRecordingPublisher
+            .sink { [weak self] isRecording in
+                if isRecording == true {
+                    self?.model?.startRecording()
+                } else {
+                    self?.model?.stopRecording()
+                }
+            }
+            .store(in: &cancellables)
+
+        recordManager.recordedDurationPublisher
+            .sink { [weak self] recordedDuration in
+                self?.model?.updateRecordingTime(recordedDuration)
+            }
+            .store(in: &cancellables)
+
+        recordManager.amplitudePublisher
+            .sink { [weak self] amplitude in
+                self?.model?.appendAmplitdue(amplitude)
```

Comment:
모델 인텐드 동일하게 오타난 것 같습니당
```suggestion
                self?.model?.appendAmplitude(amplitude)
```